apps/meteor/server/services/reads/service.ts

```diff
@@ -0,0 +1,40 @@
+import { Reads } from '@rocket.chat/models';
+import type { ISubscription } from '@rocket.chat/core-typings';
+import { ServiceClassInternal } from '@rocket.chat/core-services';
+import type { IReadsService } from '@rocket.chat/core-services';
+
+import { Messages, Subscriptions } from '../../../app/models/server';
+import { ReadReceipt } from '../../../imports/message-read-receipt/server/lib/ReadReceipt';
+
+export class ReadsService extends ServiceClassInternal implements IReadsService {
+	protected name = 'reads';
+
+	async readThread(userId: string, tmid: string): Promise<void> {
+		const read = await Reads.findOneByUserIdAndThreadId(userId, tmid);
+
+		await Reads.updateReadTimestampByUserIdAndThreadId(userId, tmid);
+
+		const threadMessage = Messages.findOneById(tmid, { projection: { ts: 1, tlm: 1, rid: 1 } });
+		if (!threadMessage || !threadMessage.tlm) {
+			return;
+		}
+
+		ReadReceipt.storeThreadMessagesReadReceipts(tmid, userId, read?.ls || threadMessage.ts);
+
+		// doesn't mark as read if not all room members have read the thread
+		const subscriptions = await Subscriptions.findByRoomId(threadMessage.rid, {
```

Comment:
this query should filter out archived subscriptions.. please remember to create a test case for this scenario

Reply:
Fixed! I'll now add some tests for this